crates/cheatcodes/src/inspector.rs

```diff
@@ -1196,9 +1196,14 @@ impl<DB: DatabaseExt + Send> Inspector<DB> for Cheatcodes {
             info!("running call in zk vm {:#?}", call);
             let persisted_factory_deps = self.persisted_factory_deps.clone();
 
+            let arc_recorded_reads = Arc::new(Mutex::new(Some(HashMap::new())));
+            let arc_recorded_writes = Arc::new(Mutex::new(Some(HashMap::new())));
+
             let ccx = foundry_zksync_core::vm::CheatcodeTracerContext {
                 mocked_calls: self.mocked_calls.clone(),
                 expected_calls: Some(&mut self.expected_calls),
+                recorded_reads: Arc::clone(&arc_recorded_reads),
```

Comment:
You can pass `Some(&mut self.accesses)` directly and update it within the `inner()` like we do for `expected_calls` - we don't need `Arc<Mutex<_>>` in this case.

Reply:
As we are handling reads and writes in different places I didn't want to use RecordAccess already defined struct but I can do it if it's more clear.